properties/base/union.py

```diff
@@ -160,14 +161,32 @@ def _unused_default_warning(self):
                 warn('Union prop default ignored: {}'.format(prop.default),
                      RuntimeWarning)
 
-    def validate(self, instance, value):
-        """Check if value is a valid type of one of the Union props"""
+    def _try_prop_method(self, instance, value, method_name):
+        """Helper method to perform a method on each of the union props
+
+        This method gathers all errors and returns them at the end
+        if the method on each of the props fails.
+        """
+        error_messages = []
         for prop in self.props:
             try:
-                return prop.validate(instance, value)
-            except GENERIC_ERRORS:
-                continue
-        self.error(instance, value)
+                return getattr(prop, method_name)(instance, value)
+            except GENERIC_ERRORS as err:
+                if hasattr(err, 'error_tuples'):
+                    error_messages += [
+                        err_tup.message for err_tup in err.error_tuples
+                    ]
+        if error_messages:
+            extra = 'Possible explanation:'
+            for message in error_messages:
+                extra += '\n    - {}'.format(message)
```

Comment:
perhaps i'm missing something as to why we need to re-raise with a merged string. 

what about defining a `__str__` and `__unicode__` and `__repr__` metthod on the custom exception classed. This these methods (one of them) merging messages in a `error_tuples` list.

Reply:
Yeah, this is great. Feeds back into having a structured representation rather than just a somewhat arbitrary string...